From 8b06054e74df502e3f89c6290fcfd1689d735e94 Mon Sep 17 00:00:00 2001
From: Sarika Sharma <quic_sarishar@quicinc.com>
Date: Mon, 7 Oct 2024 19:39:17 +0530
Subject: [PATCH] wifi: ath12k: Fix KW issue bssid NULL dereferencing

Currently, when retrieving the BSSID value using the function
ieee80211_get_bssid(), it may return NULL.
Later, in ieee80211_rx_h_sta_process(), this value is used to
check the equality of Ethernet addresses with the function
ether_addr_equal().
Since ether_addr_equal() expects two valid addresses, this could
lead to a segmentation fault or other undefined behavior.

Hence, to prevent this add a null check for the BSSID.

Signed-off-by: Sarika Sharma <quic_sarishar@quicinc.com>
---
 net/mac80211/rx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/mac80211/rx.c b/net/mac80211/rx.c
index 660458f..3fd1e0f 100644
--- a/net/mac80211/rx.c
+++ b/net/mac80211/rx.c
@@ -1862,7 +1862,7 @@ ieee80211_rx_h_sta_process(struct ieee80211_rx_data *rx)
 	if (rx->sdata->vif.type == NL80211_IFTYPE_ADHOC) {
 		u8 *bssid = ieee80211_get_bssid(hdr, rx->skb->len,
 						NL80211_IFTYPE_ADHOC);
-		if (ether_addr_equal(bssid, rx->sdata->u.ibss.bssid) &&
+		if (bssid && ether_addr_equal(bssid, rx->sdata->u.ibss.bssid) &&
 		    test_sta_flag(sta, WLAN_STA_AUTHORIZED)) {
 			link_sta->rx_stats.last_rx = jiffies;
 			if (ieee80211_is_data_present(hdr->frame_control) &&
-- 
2.34.1

